src/server/node_manager.c

```diff
@@ -6839,6 +6839,51 @@ set_nodes(void *pobj, int objtype, char *execvnod_in, char **execvnod_out, char
 	return 0;
 }
 
+void
+disassociate_job_from_moms(job *pjob, struct pbsnode *pnode)
+{
+	int i;
+	int j;
+	int special_case = 0;
+	mom_svrinfo_t *psvrmom;
+	
+	if (pnode == NULL)
+		return;
+			
+	for (i = 0; i < pnode->nd_nummoms; i++) {	
+		if (pnode->nd_moms[i] == NULL)	
+			continue;	
+		psvrmom = (mom_svrinfo_t *)(pnode->nd_moms[i]->mi_data);	
+
+		for (j=0; j<psvrmom->msr_jbinxsz; j++) {	
+			if (psvrmom->msr_jobindx[j] == pjob) {	
+				psvrmom->msr_jobindx[j] = NULL;	
+				if (is_called_by_job_purge)
+					special_case = 1;
+			}	
+		}
+		
+		if (special_case) {	
```

Comment:
Do we really need this now? I think we can get rid of this

Reply:
This portion of code was already present for some other bug debugging, If the developer who added this code felt that it solved their purpose, let them remove it. That's why I preserved this change as it is. It is out of the scope of job disassociation from node and mom PR.

Reply:
This code was added ~3 years back and I think it served its purpose. But still, let's hear from the author of this code, @nithinj Do we need this special case of debug message anymore? If not then I would like to get rid of it.

Reply:
As I had mentioned earlier, whether we decide to remove this bug specific debugging code or not, that should be handled separately by the original developer in separate PR. I don't want to delay this PR merge further with the "out of scope" code changes.

Reply:
This is added at least two releases before. If it did not have helped so far, then I don't see any point in keeping it.